decidim-proposals/app/commands/decidim/proposals/admin/update_proposal.rb

```diff
@@ -28,6 +28,11 @@ def initialize(form, proposal)
         def call
           return broadcast(:invalid) if form.invalid?
 
+          if defined?(params)
+            attachment = params[:proposal][:attachment]
+            delete_attachment(attachment)
+          end
+
```

Comment:
hi @LauraJaimeC , as @ahukkanen commented we should use the `form` object here instead of accessing the `params` which, by the way, I don't know how's that they are reachable.

Reply:
hi @tramuntanal, can you review last commit with refactor? Thanks!!

Reply:
hello @tramuntanal, can you review the refactor? thanks!